controllers/admin/AdminCarriersController.php

```diff
@@ -649,6 +649,33 @@ public function getList($id_lang, $order_by = null, $order_way = null, $start =
                 $this->_list[$key]['name'] = Carrier::getCarrierNameFromShopName();
             }
         }
+
+        // Add specific sort for naming
+        if ($this->_orderBy == 'name') {
+            usort($this->_list, [$this, 'sortListByCarrierName']);
+        }
+    }
+
+    /**
+     * Sort arrays based on the carrier name
+     *
+     * @param array $a The first array to compare
+     * @param array $b The second array to compare
+     *
+     * @return int 0 if they are equal. If the order way is ASC, -1 for lower and 1 for greater, otherwise, it's the opposite
+     */
+    private function sortListByCarrierName(array $a, array $b): int
+    {
+        $aName = isset($a['name']) ? strtolower($a['name']) : '';
+        $bName = isset($b['name']) ? strtolower($b['name']) : '';
+        if ($aName == $bName) {
+            return 0;
+        }
+        if ($this->_orderWay == 'ASC') {
+            return ($aName < $bName) ? -1 : 1;
```

Comment:
```suggestion
            return $aName <=> $bName;
```

Seee https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.spaceship-op

Reply:
Alternatively, remove the strtolower above and
```suggestion
            return strnatcasecmp($a['name'] ?? '', $b['name'] ?? '');
```

Reply:
Wow. I completly forgot that. I love it.